overlord/state/change.go

```diff
@@ -422,13 +423,35 @@ func (c *Change) Status() Status {
 	panic(fmt.Sprintf("internal error: cannot process change status: %v", statusStats))
 }
 
+func (c *Change) notifyStatusChange(new Status) {
+	if c.lastObservedStatus == new {
+		return
+	}
+	c.state.notifyChangeStatusChangedObservers(c, c.lastObservedStatus, new)
+	c.lastObservedStatus = new
+}
+
 // SetStatus sets the change status, overriding the default behavior (see Status method).
 func (c *Change) SetStatus(s Status) {
 	c.state.writing()
-	c.status = s
-	if s.Ready() {
-		c.markReady()
+
+	if s != DefaultStatus {
+		if s.Ready() {
+			c.markReady()
+		}
+		c.notifyStatusChange(s)
+		c.status = s
```

Comment:
why this order? shouldn 't we set c.status = s first like below?

Reply:
Changed this